app/models/restaurant.rb

```diff
@@ -16,6 +16,28 @@ class Restaurant < ActiveRecord::Base
   validates :city, presence: true,
                       length: { minimum: 3, maximum: 40 }
 
+  def average_rating
+    reviews.average(:rating).to_f.round(1)
+  end
+
+  def update_rating_cache
+    self.rating_cache = average_rating
+    self.save!
+  end
+
+  def get_rating
+    if rating_cache == 0.0
+      if reviews.none?
+        "No Rating yet. Be the first!"
+      else
+        update_rating_cache
+        rating_cache
+      end
+    else
+      rating_cache
+    end
+  end
```

Comment:
Can you extract the string out of this method? Ruby methods are a bit clearer when you are returning one type of class. Here you are returning a string or `rating_cache` depending on inputs.